commands/apps.go

```diff
@@ -331,59 +373,57 @@ func RunAppsCreateDeployment(c *CmdConfig) error {
 	return c.Display(displayers.Deployments{deployment})
 }
 
-// waitForAppDeploymentRunning waits for a app deployment to be running.
-func waitForAppDeploymentRunning(apps do.AppsService, appID string, deploymentID string) (*godo.Deployment, error) {
-	failCount := 0
+func waitForActiveDeployment(apps do.AppsService, appID string, deploymentID string) error {
+	tickerInterval := 10 //10s
+	timeout := 1800      //1800s, 30min
+	n := 0
 	printNewLineSet := false
-	for i := 0; ; i++ {
-		if i != 0 {
+
+	ticker := time.NewTicker(time.Duration(tickerInterval) * time.Second)
+	for range ticker.C {
+		if n*tickerInterval > timeout {
+			ticker.Stop()
+			break
+		}
+
+		if n != 0 {
 			fmt.Fprint(os.Stderr, ".")
 			if !printNewLineSet {
 				printNewLineSet = true
 				defer fmt.Fprintln(os.Stderr)
 			}
 		}
 
-		deployment, err := apps.GetDeployment(appID, deploymentID)
-		if err == nil {
-			failCount = 0
-		} else {
-			// Allow for transient API failures
-			failCount++
-			if failCount >= maxAPIFailures {
-				return nil, err
+		if deploymentID == "" {
+			app, err := apps.Get(appID)
+			if err != nil {
+				return fmt.Errorf(("Error trying to read app deployment state: " + err.Error()))
```

Comment:
```suggestion
				return err
```

We could just return the `err` directly here and handle the messaging further up the stack. Looking at how the error is checked in `RunAppsCreateDeployment`, the output becomes a bit repetitive: 

     App deployment couldn't enter `running` state: Error trying to read app deployment state: <actual error>